fixham/internal/release/releasenotes.go

```diff
@@ -0,0 +1,228 @@
+package release
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/fixham/internal/utils"
+	"github.com/projectcalico/calico/fixham/internal/version"
+)
+
+const (
+	releaseNoteLabel        = "release-note-required"
+	releaseNotesFolder      = "release-notes"
+	releaseNoteTemplatePath = "/fixham/assets/release-note.md.tmpl"
+)
+
+var (
+	repos = []string{"calico", "bird"}
+)
+
+// ReleaseNoteIssueData represents the data for an release note issue
+type ReleaseNoteIssueData struct {
+	ID     int
+	Note   string
+	Repo   string
+	URL    string
+	Author string
+}
+
+// ReleaseNoteData represents the data for release notes
+type ReleaseNoteData struct {
+	Date         string
+	BugFixes     []*ReleaseNoteIssueData
+	OtherChanges []*ReleaseNoteIssueData
+}
+
+// milestoneNumber returns the milestone number for a given milestone
+func milestoneNumber(client *github.Client, owner, repo, milestone string) (int, error) {
+	for {
+		milestones, resp, err := client.Issues.ListMilestones(context.Background(), owner, repo, nil)
+		if err != nil {
+			return -1, err
+		}
+		for _, m := range milestones {
+			if m.GetTitle() == milestone {
+				return m.GetNumber(), nil
+			}
+		}
+		if resp.NextPage == 0 {
```

Comment:
I don't think we're ever passing `NextPage` to the `ListMilestones` call, so we're not actually getting pagination here. 

If the # of milestones is ever more than a single page, we'll loop forever always listing the first page.

We need to pass a `MilestoneListOption` to the ListMilestones call. Ideally with a few things:

- We want to filter to "open" milestones.
- We want to pass the NextPage from the previous call, if there was one. 

https://github.com/google/go-github/blob/master/github/issues_milestones.go#L39-L53

Reply:
Looks like you have it correct in `prIssuesByRepo`, so just modelling after that would be good.